compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/PlatformLayersComposeScene.skiko.kt

```diff
@@ -0,0 +1,198 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.scene
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Composition
+import androidx.compose.runtime.CompositionContext
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.InternalComposeUiApi
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Canvas
+import androidx.compose.ui.input.key.KeyEvent
+import androidx.compose.ui.input.pointer.PointerInputEvent
+import androidx.compose.ui.node.LayoutNode
+import androidx.compose.ui.node.RootNodeOwner
+import androidx.compose.ui.platform.setContent
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.LayoutDirection
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.Dispatchers
+
+/**
+ * Constructs a single-layer [ComposeScene] using the specified parameters. It utilizes
+ * [ComposeSceneContext.createPlatformLayer] to position a new [LayoutNode] tree.
+ *
+ * After [ComposeScene] will no longer needed, you should call [ComposeScene.close] method, so
+ * all resources and subscriptions will be properly closed. Otherwise, there can be a memory leak.
+ *
+ * @param density Initial density of the content which will be used to convert [Dp] units.
+ * @param layoutDirection Initial layout direction of the content.
+ * @param size The size of the [ComposeScene]. Default value is `null`, which means the size will be
+ * determined by the content.
+ * @param coroutineContext Context which will be used to launch effects ([LaunchedEffect],
+ * [rememberCoroutineScope]) and run recompositions.
+ * @param composeSceneContext The context to share resources between multiple scenes and provide
+ * a way for platform interaction.
+ * @param invalidate The function to be called when the content need to be recomposed or
+ * re-rendered. If you draw your content using [ComposeScene.render] method, in this callback you
+ * should schedule the next [ComposeScene.render] in your rendering loop.
+ * @return The created [ComposeScene].
+ *
+ * @see ComposeScene
+ */
+@InternalComposeUiApi
+fun PlatformLayersComposeScene(
+    density: Density = Density(1f),
+    layoutDirection: LayoutDirection = LayoutDirection.Ltr,
+    size: IntSize? = null,
+    coroutineContext: CoroutineContext = Dispatchers.Unconfined,
```

Comment:
Now would be a good time to change this default here and on `CanvasLayersComposeScene`

https://github.com/JetBrains/compose-multiplatform/issues/4799

Reply:
Good point, thanks for reminder

Reply:
After discussions with @igordmn and @m-sasha we didn't agree what should be used as a default value here instead. So it was postponed and moved out of the scope of this PR - we'll have some room to do that before 1.7 stable without breaking compatibility